app/controllers/login_controller.rb

```diff
@@ -13,10 +14,9 @@ def attempt_login_user
     error = nil
     if user.nil?
       error = 'Invalid username'
-    elsif !user.password == @request.params['password']
+    elsif BCrypt::Password.new(user.password) != @request.params['password']
       error = 'Invalid password'
```

Comment:
This also fixes the fact that our original non-functional password check didn't actually check the hash, but the plain password instead. The problem only came to light after we started checking for password equality after the other bug was fixed.